### PR TITLE
Add sql proj schema compare for dacpac

### PR DESCRIPTION
**Pull request by kisantia**
This adds basic support for schema compare for sql database projects by building the dacpac of the project and setting that as the source when the schema compare is opened
Screen Shot 2020-05-13 at 7 25 59 PM

Error if schema compare extension isn't installed:
Screen Shot 2020-05-13 at 5 08 36 PM

### CODE REVIEW

File: extensions/schema-compare/src/schemaCompareMainWindow.ts [hunk 1]
```diff
@@ -91,6 +96,16 @@ export class SchemaCompareMainWindow {
 				packageFilePath: '',
 				connectionDetails: undefined
 			};
+		} else if (sourceDacpac) {
```
Comment (udeeshagautam) [inline extensions/schema-compare/src/schemaCompareMainWindow.ts:99]: in favor of CC - can we add a test case for checking endpoint creation for different inputs.. not sure if calling start() in UT will be possible but something to check

File: extensions/sql-database-projects/src/controllers/projectController.ts [hunk 2]
```diff
@@ -138,6 +138,21 @@ export class ProjectsController {
...
+		if (this.apiWrapper.getExtension(constants.schemaCompareExtensionId)) {
+			// build project
+			await this.buildProject(treeNode);
+
+			// start schema compare with the dacpac produced from build
+			const project = this.getProjectContextFromTreeNode(treeNode);
+			const dacpacPath = path.join(project.projectFolderPath, 'bin', 'Debug', `${project.projectFileName}.dacpac`);
+			this.apiWrapper.executeCommand('schemaCompare.start', dacpacPath);
+		} else {
+			this.apiWrapper.showErrorMessage(constants.schemaCompareNotInstalled);
```
Comment (udeeshagautam) [inline extensions/sql-database-projects/src/controllers/projectController.ts:152]: not necessary for this PR but can we have a "do you want to install - yes/no" and opening the extn. viewlet with SC searched?. Is the second part (opening Extn view with SC) possible?

Comment (Benjin) [reply extensions/sql-database-projects/src/controllers/projectController.ts:152]: I like this idea!  Help reduce friction for using the feature

File: extensions/schema-compare/src/schemaCompareMainWindow.ts (hunk 1 above)
Comment (kisantia) [reply extensions/schema-compare/src/schemaCompareMainWindow.ts:99]: added tests for the 3 different contexts schema compare can start with

File: extensions/sql-database-projects/src/controllers/projectController.ts (hunk 2 above)
Comment (kisantia) [reply extensions/sql-database-projects/src/controllers/projectController.ts:158]: opened issue to track this #10415

**Pull request merged**